### PR TITLE
feat: make clicking a closing brace inlay hint go to the opening brace

### PR DESCRIPTION
**Pull request by jonas-schievink**

### CODE REVIEW

File: crates/rust-analyzer/src/to_proto.rs [hunk 1]
```diff
@@ -459,15 +467,9 @@ pub(crate) fn inlay_hint(
             | InlayKind::ImplicitReborrowHint
             | InlayKind::TypeHint
             | InlayKind::ClosingBraceHint => false,
-            InlayKind::BindingModeHint => inlay_hint.label != "&",
+            InlayKind::BindingModeHint => inlay_hint.label.to_string() != "&",
```
Comment (Veykril) [inline crates/rust-analyzer/src/to_proto.rs:470]:
```suggestion
            InlayKind::BindingModeHint => inlay_hint.label.as_simple_str() != Some("&"),
```

**Review by Veykril — APPROVED** [review]

**Comment by lnicola** [comment]
I couldn't spot why, but this only works for items, not `impl` blocks.

https://user-images.githubusercontent.com/308347/189642703-3b16578b-f500-4925-b6a2-0ca25b4c119f.mp4

**Comment by Veykril** [comment]
It's only implemented for things that have a name as it jumps to the name, not the opening brace actually

**Comment by lnicola** [comment]
Yeah, that's a bit unexpected, since the closing brace hint shows up for  `impl Foo`.

**Comment by jonas-schievink** [comment]
Somewhat cursed idea: what if we make `textDocument/definition` on a closing brace return the corrsponding opening brace. That would make this feature work as expected, even for `impl`s.

**Pull request merged**